xarray/backends/pydap_.py

```diff
@@ -97,29 +100,26 @@ def open(
         user_charset=None,
     ):
 
-        if output_grid is None:
-            output_grid = True
-
-        if verify is None:
-            verify = True
-
         if timeout is None:
             from pydap.lib import DEFAULT_TIMEOUT
 
             timeout = DEFAULT_TIMEOUT
 
-        if user_charset is None:
-            user_charset = "ascii"
-
-        ds = pydap.client.open_url(
-            url=url,
-            application=application,
-            session=session,
-            output_grid=output_grid,
-            timeout=timeout,
-            verify=verify,
-            user_charset=user_charset,
-        )
+        kwargs = {
+            "url": url,
+            "application": application,
+            "session": session,
+            "output_grid": output_grid or True,
+            "timeout": timeout,
+        }
+        if Version(pydap_version) >= Version("3.3.0"):
+            kwargs.update(
+                {
+                    "verify": verify if verify is not None else True,
+                    "user_charset": user_charset or "ascii",
+                }
+            )
```

Comment:
OK then this should preserve pydap's defulat values even if they change it

```suggestion
            if verify is not None:
                kwargs.update({"verify": verify}) 
            if user_charset is not None:
                kwargs.update({"user_charset": user_charset})
```